src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -3,34 +3,88 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.Extensions.Configuration
 {
-    public class ConfigurationRoot : ConfigurationBase, IConfigurationRoot
+    public class ConfigurationRoot : IConfigurationRoot
     {
+        private IList<IConfigurationProvider> _providers;
+        private ConfigurationReloadToken _reloadToken = new ConfigurationReloadToken();
+
         public ConfigurationRoot(IList<IConfigurationProvider> providers)
-            : base(providers)
         {
             if (providers == null)
             {
                 throw new ArgumentNullException(nameof(providers));
             }
+
+            _providers = providers;
         }
 
-        public override string Path
```

Comment:
BTW, I know this is at odds with the reasons we specialized `IConfigurationRoot` and `IConfigurationSection`. I sometimes feel that maybe we went too far with that.